File: backend/src/main/java/com/runninggu/server/festival/domain/Festival.java
package com.runninggu.server.festival.domain;

import java.math.BigDecimal;
import java.time.LocalDate;

/** 한국관광공사 축제 검색 결과를 서버 축제 기능이 함께 쓰는 값이다. (SPEC §4.4·§8.3) */
public record Festival(
        String contentId,
        String name,
        LocalDate startDate,
        LocalDate endDate,
        /** KTO가 좌표를 주지 않으면 null이다. 홈 월간 축제는 좌표 없이도 노출할 수 있다. */
        BigDecimal lat,
        BigDecimal lng,
        String imageUrl,
        String address) {

    public boolean hasCoordinates() {
        return lat != null && lng != null;
    }
}

File: android/app/src/main/java/com/runninggu/app/data/model/Festival.kt
package com.runninggu.app.data.model

import java.time.LocalDate

/**
 * 홈 축제 섹션 한 건. (API 명세 §4-1 · SPEC §4.4)
 *
 * [inProgress] 는 **서버 판정을 그대로 쓴다.** 앱이 오늘 날짜로 다시 계산하면 서버와
 * 기준이 갈린다 — 배지 하나 때문에 두 벌 규칙을 만들 이유가 없다.
 */
data class Festival(
    val contentId: String,
    val name: String,
    val startDate: LocalDate?,
    val endDate: LocalDate?,
    val region: String,
    val imageUrl: String?,
    val inProgress: Boolean,
)
